pkg/daemon/update.go

```diff
@@ -1727,7 +1807,13 @@ func (dn *Daemon) updateLayeredOS(config *mcfgv1.MachineConfig) error {
 		if err := dn.InplaceUpdateViaNewContainer(newURL); err != nil {
 			return err
 		}
-	} else if err := dn.NodeUpdaterClient.RebaseLayered(newURL); err != nil {
+	}
+
+	if err := dn.prepForRHCOS9(newURL); err != nil {
```

Comment:
I'm fairly certain this is correct for the `applyLayeredOSChanges()` path. However, I'm not as certain for the `applyLegacyOSChanges()` path.

Reply:
I think `applyLegacyOSChanges()` may be fully dead code now?  cc @jkyros

Reply:
If that path is dead, I'd like to remove it in a follow-up PR. No sense in keeping it around if it's not being used.

Reply:
`applyLegacyOSChanges()` can only be reached if someone feeds us an "old format" image. I'm assuming the only way to get to 4.13 is through 4.12 version (like, we don't support someone forcing an upgrade from 4.11 to 4.13), so we can safely take that out for 4.13, which is awesome, because that was an unpleasant bit of code. :smile: